### DISCUSSION TITLE
Unable to communicate with Crazyflie using Chooser.py

### DISCUSSION THREAD

**Discussion by imrd1123** [USC-ACTLab/crazyswarm#508] [Q&A] [answered]
Environment
OS=Ubuntu 20.04 (BitcrazeVM)
ROS=Noetic
Python=3.8.10

---

Hi. I'm in trouble with installing crazyswarm.
I'm glad if you help to solve my problem.

This may be a rudimentary question.
I'm trying to connect to crazyflie using chooser.py.
When I turn on the power of one Crazyflie and run chooser.py with roscore running, there is no communication.
("Err" message appears when I click on "battery" etc.)
![image](https://user-images.githubusercontent.com/9507900/139190667-8ecf5fad-8bb9-4336-a788-76ddb954d3a2.png)

The settings in allCrazyflies.yaml are as follows.
I am planning to run it on channel 100.
![image](https://user-images.githubusercontent.com/9507900/139190466-3fc0eab9-a935-489c-9d4c-88071f851e6f.png)

Also, Crazyflie is set to "Ch.100" in cfclient.
![image](https://user-images.githubusercontent.com/9507900/139190565-f969e9b1-68fa-4f8f-8c82-d6a1138936ad.png)

The message when chooser.py is invoked is as follows.
![image](https://user-images.githubusercontent.com/9507900/139194377-215bbbee-5dbd-458e-9985-d818a05e78f8.png)

I would appreciate it if you could tell me what I am doing wrong.

**Comment by jpreiss** [comment, marked as answer]
The `id` field in `allCrazyflies.yaml` must be the last byte of the radio address in decimal. So if your CF has address `0xE7E7E7E7`, then its `id` should be `231`.

Also keep in mind you do not need to have `roscore` running to use `chooser.py`, and you cannot use `chooser.py` if you are already running `crazyswarm_server`.

**Reply by imrd1123** [reply]
Hi James

I understand about the id setting and roscore. 
I tried it by entering 231 as the `id,` and was able to communicate successfully.
Thank you very much.

However, I have a question about server.
How do I start `crazyswarm_server`?
Is it enough to execute the following code?
`source ros_ws/devel/setup.bash`
`roslaunch crazyswarm hover_swarm.launch`

When I run the code, the `rviz` screen comes up, 
but it does not seem to be connected to Crazyflie and does not appear in the 3D map.

---
↑Sorry I understand why
![image](https://user-images.githubusercontent.com/9507900/139393801-db63c7b5-5da9-4beb-a183-a9b274e956fb.png)

**Reply by whoenig** [reply]
* The launch-file starts the crazyswarm_server node
* If nothing shows up in rviz, there might be a connection problem with your external tracking. Are you using motion capture or something else?

**Reply by imrd1123** [reply]
Hi Wolfgang

Thank you for teach me.
I understand that the following code will bring up the crazyswarm_server node.
`source ros_ws/devel/setup.bash`
`roslaunch crazyswarm hover_swarm.launch`

You are right.
I am using Vicon, and was trying to communicate with it unconnected.
After connecting with the communication cable, I filled in the `motion_capture_host_name` with the correct IP address, and changed the marker name to `cf231.`
By doing so, I was able to make the `cf231` marker appear in `rviz.`
Thank you very much for your help in solving this problem.

---

However, I have another problem, I want to run multiple Crazyflie from one radio at the same time.
I have configured the settings in allCrazyflies.yaml as follows, but I can't get them to communicate properly.
What am I doing wrong?

```
# ros_ws/src/crazyswarm/launch/allCrazyflies.yaml
crazyflies:
  - id: 231
    channel: 1
    initialPosition: [1.5, 1.5, 0.0]
    type: default
  - id: 231
    channel: 2
    initialPosition: [1.5, 1.0, 0.0]
    type: default

```
I'm sorry to bother you, but I would appreciate it if you could tell me. Thank you very much.

**Reply by whoenig** [reply]
Every robot needs to have a unique address. To use a single radio, the channel needs to be identical. If you use multiple channels, they need to be spaced by at least 2 apart. In most scenarios, channels 60-100 are good.

**Reply by imrd1123** [reply]
Thank you for your answer.
I understand a little about communication rules.
I'll try to set channels spaced 10 (Ch100, Ch110).

**Comment by imrd1123** [comment]
I'm sorry, I don't know much about communication yet.
I am trying to run two Crazyflie units on a single CrazyradioPA.
I am thinking of using Single marker as the operation method. The devices are set up as follows
by `cfclient`
　CF_1: Address[0xE7E7E710], Ch.110
　CF_2: Address[0xE7E7E720], Ch.120
`allcrazyflie.yaml`
　- id:10
　  channel:110
　  initialPosition:[0.0, 0.2, 0.0].
　  type:default
　- id:20
　  channel:120
　  initialPosition:[0.0, -0.2, 0.0].
　  type:default

I started chooser.py in this state, but it could not communicate.
I have set the `Address` and the `id` to be the same, but what am I doing wrong?
Please let me know.

**Reply by whoenig** [reply]
The address is hexadecimal; 10 = "0x0A" and 20 = "0x14", i.e., the first address should be 0xE7E7E7E70A, second should be 0xE7E7E7E714.

**Reply by imrd1123** [reply]
I did not notice the fundamental mistake. I apologize for this.
I solved the problem successfully and was able to fly the two Crazyflies at the same time. 
Thank you very much for answering so many questions.
